File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
class Raindrops {
    companion object {
        fun convert(n: Int) = buildString {
            if ((n % 3) == 0) append("Pling")
            if ((n % 5) == 0) append("Plang")
            if ((n % 7) == 0) append("Plong")
            if (isBlank()) append(n)
        }

        // works but not as nice as buildString()
        fun convert2(i: Int): String {
            fun checkFactor(i: Int, factor: Int, word: String) = if ((i % factor) == 0) word else ""
            val output = checkFactor(i, 3, "Pling") + checkFactor(i, 5, "Plang") + checkFactor(i, 7, "Plong")
            return if (output.isEmpty()) i.toString() else output
        }

    }
}

File: go/raindrops/raindrops.go
package raindrops

import "fmt"

// Convert raindrops to funny string
func Convert(n int) (out string) {
	if (n % 3) == 0 {
		out += "Pling"
	}
	if (n % 5) == 0 {
		out += "Plang"
	}
	if (n % 7) == 0 {
		out += "Plong"
	}
	if out == "" {
		out += fmt.Sprintf("%d", n)
	}
	return
}

/*
	fun convert(n: Int) = buildString {
		if ((n % 3) == 0) append("Pling")
		if ((n % 5) == 0) append("Plang")
		if ((n % 7) == 0) append("Plong")
		if (isBlank()) append(n)
	}
*/
